File: FSH_wanxi/src/com/ict/servlet/Switch.java
package com.ict.servlet;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

import com.ict.java_bean.SendOnly;

/**
 * Created with IntelliJ IDEA.
 * User: StevenH
 * Date: 13-10-11
 * Time: 下午4:33
 * 此servlet负责控制switch节点的开关状态
 * To change this template use File | Settings | File Templates.
 */
@WebServlet(name = "Switch")
public class Switch extends HttpServlet {
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        request.setCharacterEncoding("UTF-8");
        String mac = request.getParameter("mac");
        int state = Integer.parseInt(request.getParameter("state"));
        String conState="Mac "+mac;
        if(state==0){
            conState+=" 0 CmdNrm Switch_On";
//            state=1;
        }else if(state==1){
            conState+=" 1 CmdNrm Switch_Off";
//            state=0;
        }
        System.out.println("******************操作语句******************");
        System.out.println("22222222222222222222222222222222222222222222222");
        System.out.println(conState);
        System.out.println("******************语句结束******************");

        SendOnly send=new SendOnly(conState);
        send.run();

        response.getWriter().println("{ mac : \"" + mac + "\",state:" + state + "}");
    }

    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
        doPost(request,response);
    }
}

File: FSH_wanxi/web/js/Switch.js
/**
 * Created with IntelliJ IDEA.
 * User: StevenH
 * Date: 13-10-14
 * Time: 下午3:24
 * To change this template use File | Settings | File Templates.
 */
function swhClick(){
    $.post("Switch.do",{
        mac : $("#param1").val(),
        state: $("#param2").val()
    },function(data,textStatus){
        /*var state =data.state;
        var txtHtml;
            txtHtml = "<div>设备状态:&nbsp";
            switch (data.state){
                case 0 :
                    txtHtml += "关<br><br>";
                    txtHtml += "控制:&nbsp&nbsp&nbsp"
                        + "<button class='switch' onclick='swhClick()'>&nbsp&nbsp开&nbsp&nbsp</button>";
                    // 加入隐藏域传递值
                    txtHtml += "<input type='hidden' id='param1' value='"
                        + data.mac + "'/>";
                    txtHtml += "<input type='hidden' id='param2' value='"
                        + data.state + "'/>";
                    break;
                case 1 :
                    txtHtml += "开<br><br>";
                    txtHtml += "控制:&nbsp&nbsp&nbsp"
                        + "<button class='switch' onclick='swhClick()'>&nbsp&nbsp关&nbsp&nbsp</button>";
                    // 加入隐藏域传递值
                    txtHtml += "<input type='hidden' id='param1' value='"
                        + data.mac + "'/>";
                    txtHtml += "<input type='hidden' id='param2' value='"
                        + data.state + "'/>";
                    break;
            }
            txtHtml += "</div>";
            $("#resState").html(txtHtml);*/
    },"json");
    setTimeout("statePrint()",1000);
}
function statePrint(){
    $.get("Type.do", {
        name : encodeURI($("#param3").val())
    }, function(data, textStatus) {
        if(data.mac!=null){
            var txtHtml = "<div>设备状态:&nbsp";
            switch (data.state) {
                case 0 :
                    txtHtml += "关<br><br>";
                    txtHtml += "控制:&nbsp&nbsp&nbsp"
                        + "<button class='switch' onclick='swhClick()'>&nbsp&nbsp开&nbsp&nbsp</button>";
                    // 加入隐藏域传递值
                    txtHtml += "<input type='hidden' id='param1' value='"
                        + data.mac + "'/>";
                    txtHtml += "<input type='hidden' id='param2' value='"
                        + data.state + "'/>";
                    break;
                case 1 :
                    txtHtml += "开<br><br>";
                    txtHtml += "控制:&nbsp&nbsp&nbsp"
                        + "<button class='switch' onclick='swhClick()'>&nbsp&nbsp关&nbsp&nbsp</button>";
                    // 加入隐藏域传递值
                    txtHtml += "<input type='hidden' id='param1' value='"
                        + data.mac + "'/>";
                    txtHtml += "<input type='hidden' id='param2' value='"
                        + data.state + "'/>";
                    break;
                default :
                    txtHtml += "未使用<br><br>";
                    txtHtml += "控制:&nbsp&nbsp&nbsp"
                        + "<button class='switch' onclick='swhClick()'>&nbsp&nbsp开&nbsp&nbsp</button>";
                    // 加入隐藏域传递值
                    txtHtml += "<input type='hidden' id='param1' value='"
                        + data.mac + "'/>";
                    txtHtml += "<input type='hidden' id='param2' value='"
                        + 1 + "'/>";
                    break;
            }
            txtHtml += "</div>";
            $("#resState").html(txtHtml);
        }else{
            var txtHtml2 = "<div>该设备还没有开启！</div>";
            $("#resState").html(txtHtml2);
        }
    },"json");
}